web/auth/user_from_auth.ex

```diff
@@ -102,6 +102,19 @@ defmodule PhoenixGuardian.UserFromAuth do
     end
   end
 
+  defp auth_and_validate(%{provider: service} = auth, repo)  when service in [:google, :facebook] do
+    case repo.get_by(Authorization, uid: uid_from_auth(auth), provider: to_string(auth.provider)) do
+      nil -> {:error, :not_found}
+      authorization ->
+        if authorization.uid == uid_from_auth(auth) do
+          authorization
+        else
+          {:error, :"#{auth.provider} uid mismatch"}
```

Comment:
I think here we either want:

* `:uid_mismatch`
* `:"#{auth.provider}_uid_mismatch"`
* `"#{auth.provider} uid mismatch

Creating an atom here with spaces is weird.